test/rules/no-access-state-in-setstate/test.tsx.lint

```diff
@@ -0,0 +1,68 @@
+class SomeReactComponent extends React.Component {
+
+    someClassFunction() {
+
+        this.fooBar({
+            foo: this.state.foo
+        });
+
+        this.setState({
+            foo: "foo",
+            bar: this.barz
+        });
+
+        this.setState(
+            {
+                foo: "foo"
+            },
+            () => this.fooBar(this.state.foo);
+        );
+
+        this.setState(prevState => ({
+            foo: !prevState.foo
+        }));
+
+        this.setState((prevState, currentProps) => ({
+            foo: !prevState.foo,
+            bar: currentProps.bar
+        }));
+
+        this.setState({
+        ~~~~~~~~~~~~~~~
+            foo: !this.state.foo
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+        });
+~~~~~~~~~~ [0]
+
+        this.setState({
+        ~~~~~~~~~~~~~~~
+            foo: this.fooBar(this.state.foo)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+        });
+~~~~~~~~~~ [0]
+
+        this.setState((prevState, currentProps) => ({
+        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+            foo: !this.state.foo,
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+            bar: currentProps.bar
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+        }));
+~~~~~~~~~~~ [0]
+
+        this.setState((prevState, currentProps) => {
+        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+            this.fooBar(this.state.foo);
```

Comment:
Very good idea, will do that!

Reply:
Okay, unfortunately it might be a little difficult. In my now updated code, you'd have to copy the `callbackForEachChildInSetStateArgument` function, for checking a callback and adding another failure text to the node. At least that's the only way I can think of right now, but I don't have that much experience with TSLint yet :-)